src/Symfony/Component/Config/Definition/Builder/NodeFinder.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Definition\Builder;
+
+/**
+ * This class finds nodes in a node tree.
+ *
+ * @author Jan Schädlich <jan.schaedlich@sensiolabs.de>
+ */
+class NodeFinder
+{
+    /**
+     * Finds a node defined by $nodePath within the given $rootNode.
+     *
+     * @param string $nodePath The path of the node to find. e.g "doctrine.orm.mappings"
+     */
+    public function find(string $nodePath, NodeDefinition $rootNode): NodeDefinition
+    {
+        if (!$rootNode instanceof ArrayNodeDefinition) {
```

Comment:
to me this shows it could also be solved using `ArrayNodeDefinition::find()`, which would avoid the additional getters right?

Reply:
Sounds good to me. Then we also don't need this class at all.